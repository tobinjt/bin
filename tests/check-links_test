#!/bin/bash

set -e -f -u -o pipefail

source ../check-links

wget() {
  cat > wget.log <<WGET
# Successful request
--2019-12-12 22:45:34--  https://www.johntobin.ie/
HTTP request sent, awaiting response... 200 OK
# Unsuccessful request
--2019-12-12 22:45:34--  https://www.johntobin.ie/unsuccessful
HTTP request sent, awaiting response... 404 Not found
# Unsuccessful request, succeeds on retry
--2019-12-12 22:45:34--  https://www.johntobin.ie/succeed-on-retry
HTTP request sent, awaiting response... 401 Something went wrong
--2019-12-12 22:45:34--  https://www.johntobin.ie/succeed-on-retry
HTTP request sent, awaiting response... 200 OK
WGET
}

# main() should fail because of the errors in the wget output.
(main "https://www.johntobin.ie/" || printf "main failed as expected\\n" >&2) \
  | sed -e '/for further investigation/d'
