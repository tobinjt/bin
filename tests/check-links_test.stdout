Problems found:
--2019-12-12 22:45:34--  https://www.johntobin.ie/unsuccessful HTTP request sent, awaiting response... 404 Not found
